pkgs/by-name/sp/spades/package.nix

```
@@ -2,33 +2,36 @@
   lib,
   stdenv,
   fetchurl,
-  zlib,
+  zlib-ng,
   bzip2,
   cmake,
   python3,
+  readline,
+  ncurses
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "spades";
-  version = "3.15.5";
+  version = "4.0.0";
 
   src = fetchurl {
     url = "https://github.com/ablab/spades/releases/download/v${finalAttrs.version}/SPAdes-${finalAttrs.version}.tar.gz";
-    hash = "sha256-FVw2QNVx8uexmgUDHR/Q0ZvYLfeF04hw+5O9JBsSu/o=";
+    hash = "sha256-B8AusdnZD2EaxzvdMN3CQu1RsAyKN1cYnoqBN62M+4s=";
   };
   sourceRoot = "SPAdes-${finalAttrs.version}/src";
 
-  env.CXXFLAGS = toString [
-    # GCC 13: error: 'uint32_t' does not name a type
-    "-include cstdint"
-  ];
+  patches = [ ./copytree.patch ];
```

```suggestion
  patches = [ ./copytree.patch ];
```
What is the source of this patch?
If this patch exist somewhere else, use `fetchpatch` for it and include a comment with issue/PR that originated it.

---

I made this patch and submitted it upstream... https://github.com/ablab/spades/pull/1314

---

Congrats. And thanks for fixing things in upstream. As this makes upkeep easier.
You can `fetchpatch` it then. We usually avoid directly adding patches to nixpkgs unless necessary.
Also, if upstream accepts your patch, then you may just move forward the version reference.